### PR TITLE
Rename built-in tools → native tools; register via `capabilities=[NativeTool(...)]`

### LINKED ISSUE #3609: Rename "built-in" tools to something else (server-side is one option)
### Description

"built-in" is often interpreted to mean "built into Pydantic AI"

This is a big change - for reference I implemented it in https://github.com/zby/pydantic-ai/commit/e1076d13576745cc25cb575a2c65c6517955a4f1

### References

_No response_

### PR DESCRIPTION
**Pull request by DouweM**
- Closes #3609

## Summary

Renames the "built-in tools" concept to "native tools" across the public API (classes, fields, methods, modules, message parts) and consolidates per-agent / per-call native-tool registration through the `capabilities=` parameter using `NativeTool(...)`, `WebSearch()`, `WebFetch()`, `MCP()`, `ImageGeneration()`. The 1.x `builtin_tools=` kwargs and `Builtin*` import paths continue to work with a `PydanticAIDeprecationWarning` (visible by default).

"Native" was chosen over "provider" because it fits adjectivally into prose ("supported native tools", "natively supports", "native tool search"), aligns with `NativeOutput`, and avoids tautology with the existing `Provider` class.

## Migration

| Before | After |
|---|---|
| `from pydantic_ai.builtin_tools import WebSearchTool, AbstractBuiltinTool` | `from pydantic_ai.native_tools import WebSearchTool, AbstractNativeTool` |
| `from pydantic_ai import BuiltinToolCallPart, BuiltinToolReturnPart, AgentBuiltinTool` | `from pydantic_ai import NativeToolCallPart, NativeToolReturnPart, AgentNativeTool` |
| `Agent(model, builtin_tools=[WebSearchTool()])` | `Agent(model, capabilities=[NativeTool(WebSearchTool())])` *(or `[WebSearch()]` for provider-adaptive with local fallback)* |
| `agent.run("...", builtin_tools=[...])` (also `iter`, `run_sync`, `run_stream`, `run_stream_sync`, `run_stream_events`, `to_web`) | `agent.run("...", capabilities=[NativeTool(...)])` |
| `agent.override(builtin_tools=[...])` | `agent.override(native_tools=[...])` *(kept as a first-party override surface — see note below)* |
| `BuiltinToolCallPart` / `BuiltinToolReturnPart` | `NativeToolCallPart` / `NativeToolReturnPart` *(`part_kind` literals preserved for serialization compat)* |
| `Tool.prefer_builtin`, `ModelProfile.supported_builtin_tools`, `ModelRequestParameters.builtin_tools`, `ModelResponse.builtin_tool_calls` | drop the `builtin_` prefix → `native_*` |
| `NativeOrLocalTool(builtin=...)`, `WebSearch(builtin=...)`, `WebFetch(builtin=...)`, `MCP(builtin=...)`, `ImageGeneration(builtin=...)` | use `native=` |
| `OpenAIModelSettings.openai_builtin_tools` | `openai_native_tools` |
| `GoogleModelProfile.google_supports_native_output_with_builtin_tools` | `..._with_native_tools` |

Tool subclass names (`WebSearchTool`, `XSearchTool`, `CodeExecutionTool`, `MCPServerTool`, `WebFetchTool`, `ImageGenerationTool`, `FileSearchTool`, `MemoryTool`, `UrlContextTool`) are **unchanged** — they live in `pydantic_ai.native_tools` so the namespace already disambiguates them, and prefixing would create double-native at the call site (`NativeTool(NativeWebSearchTool())`).

## Backward compatibility

Everything in the migration table also works in 1.x with a `PydanticAIDeprecationWarning` (a `UserWarning` subclass, visible by default):

- Top-level `__getattr__` shim in `pydantic_ai/__init__.py` for `BuiltinToolCallPart` / `BuiltinToolReturnPart` / `AgentBuiltinTool` so `from pydantic_ai import …` keeps working.
- `pydantic_ai.builtin_tools` remains as a `__getattr__` shim re-exporting from `pydantic_ai.native_tools` (both renamed names and unchanged tool subclasses).
- Module-level `__getattr__` aliases for renamed names in `pydantic_ai.messages`, `pydantic_ai.tools`, `pydantic_ai.capabilities`.
- Deprecated property aliases on `ModelResponse.builtin_tool_calls`, `NativeOrLocalTool.builtin`.
- `ModelRequestParameters` accepts the legacy `builtin_tools=` kwarg in its constructor AND has a `pydantic.Field(validation_alias=AliasChoices('native_tools', 'builtin_tools'))` so previously-serialized payloads still deserialize losslessly through `TypeAdapter`.
- All per-call / constructor `builtin_tools=` kwargs route through a shared helper that converts them into `NativeTool` capability wrappers; the helper is removed in v2.

Wire-format JSON keys (`builtinTools` in `ModelInfo`, `BuiltinToolInfo`, etc. — used by the chat UI protocol) are intentionally left untouched in this PR; renaming those is a separate breaking-change task on the chat UI protocol contract.

## Note about `override(native_tools=...)`

`Agent.override` is the one place where `native_tools=` survives as a first-party parameter. Reason: `override(capabilities=...)` is not currently supported (only individual configured fields like `tools`, `toolsets`, `model`, etc. can be overridden without an `AgentSpec`), so dropping `native_tools=` would leave users unable to swap just the native-tool baseline without replacing the entire capability list. May be revisited if `override(capabilities=...)` lands.

## Test plan

- [x] `tests/test_native_tools_deprecation.py`: 41 lock-in tests covering every deprecated surface (top-level + module shims, property aliases, constructor kwargs, per-call kwargs, Pydantic validation alias, both-passed edge cases).
- [x] `tests/test_capabilities.py`, `tests/test_agent.py`, `tests/test_ui.py`, `tests/test_ag_ui.py`, `tests/test_vercel_ai.py`, `tests/test_temporal.py`: existing tests migrated to the new API. All passing on 3.10–3.14 across standard / all-extras / lowest-versions / slim / evals matrices.
- [x] Snapshots regenerated for the renamed `part_kind` / event-kind literals where required.
- [x] Coverage at 100% (`strict-no-cover` clean).

### Checklist

- [ ] Any **AI generated code** has been reviewed line-by-line by the human PR author, who stands by it.
- [x] No **breaking changes** in accordance with the [version policy](https://github.com/pydantic/pydantic-ai/blob/main/docs/version-policy.md).
- [x] **PR title** is fit for the [release changelog](https://github.com/pydantic/pydantic-ai/releases).

### CODE REVIEW

File: docs/graph/beta/joins.md [hunk 1]
```diff
@@ -56,7 +56,7 @@ async def main():
 
 _(This example is complete, it can be run "as is" — you'll need to add `import asyncio; asyncio.run(main())` to run `main`)_
 
-## Built-in Reducers
```
Comment (DouweM) [inline docs/graph/beta/joins.md:59]: "Built-in" was changed to "Native" in graph and evals context, incorrectly.

File: docs/capabilities.md [hunk 2]
```diff
@@ -4,29 +4,29 @@ A capability is a reusable, composable unit of agent behavior. Instead of thread
...
+| [`WebFetch`][pydantic_ai.capabilities.WebFetch] | URL fetching — native when supported, [local fallback](common-tools.md#web-fetch-tool) with [`web-fetch` extra](install.md#slim-install) | Yes |
+| [`ImageGeneration`][pydantic_ai.capabilities.ImageGeneration] | Image generation — native when supported, subagent fallback via `fallback_model` | Yes |
+| [`MCP`][pydantic_ai.capabilities.MCP] | MCP server — native when supported, direct connection otherwise | Yes |
 | [`PrepareTools`][pydantic_ai.capabilities.PrepareTools] | Filters or modifies function [tool definitions](tools.md) per step | — |
 | [`PrepareOutputTools`][pydantic_ai.capabilities.PrepareOutputTools] | Filters or modifies [output tool][pydantic_ai.output.ToolOutput] definitions per step | — |
 | [`PrefixTools`][pydantic_ai.capabilities.PrefixTools] | Wraps a capability and prefixes its tool names | Yes |
-| [`BuiltinTool`][pydantic_ai.capabilities.BuiltinTool] | Registers a [builtin tool](builtin-tools.md) with the agent | Yes |
+| [`NativeTool`][pydantic_ai.capabilities.NativeTool] | Registers a [native tool](native-tools.md) with the agent | Yes |
```
Comment (DouweM) [inline docs/capabilities.md:29]: Do we have a redirect setup from `builtin-tools.md` to `native-tools.md`? That may require work in https://github.com/pydantic/unified-docs, check that out

File: docs/CLAUDE.md
Comment (DouweM) [inline docs/CLAUDE.md]: This was a symlink, that you've turned into a concrete file! Make it a symlink again, and update the source AGENTS.md, not the symlink CLAUDE.md

File: docs/evals.md [hunk 3]
```diff
@@ -200,7 +200,7 @@ _(This example is complete, it can be run "as is")_
 Learn more:
 
 - [Evaluators Overview](evals/evaluators/overview.md) - When to use different types
-- [Built-in Evaluators](evals/evaluators/built-in.md) - Complete reference
+- [Native Evaluators](evals/evaluators/built-in.md) - Complete reference
```
Comment (DouweM) [inline docs/evals.md]: As mentioned, shouldn't have been renamed.

File: pydantic_ai_slim/pydantic_ai/_warnings.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+from __future__ import annotations
+
+
+class PydanticAIDeprecationWarning(UserWarning):
+    """Warning emitted when a deprecated Pydantic AI API is used.
+
+    Inherits from `UserWarning` instead of `DeprecationWarning` so that
+    deprecations are visible by default at runtime, following the approach
+    described in https://sethmlarson.dev/deprecations-via-warnings-dont-work-for-python-libraries.
```
Comment (DouweM) [inline pydantic_ai_slim/pydantic_ai/_warnings.py:9]: @dsfaccini FYI, I'm planning to merge this one very shortly and your PRs may want to use this too!
